### ISSUE TITLE
Bug ResourceRelationshipGuesser::guessResource

### ISSUE THREAD

**Issue by emtudo** [laravel/nova-issues#793]
I'm using a different Laravel framework, and I made the service call in every domain I need. Then the following happens:

I have the following files:
App \ Domains \ Projects \ Resources \ Domain
App \ Domains \ Projects \ Resources \ File

Both make a belongsTo with the file:
App \ Domains \ Projects \ Resources \ Project

So when I use Laravel \ New \ Fields \ BelongsTo
Inside the "Domain" and "File", the file works perfectly.

Domain already occurs an error, the method instead of returning to BelongsTo:
App \ Domains \ Projects \ Resources \ Project it is deploying: App \ Projects \ Projects \ Resources \ Project

In short, it replaces all names, rather than replacing only the last.

In my case I solved the third parameter to inform the resource, but my contribution is here.

**Comment by davidhemphill** [comment]
I cannot makes sense of what your bug report. Please post the actual code and a description of the bug and I'll reopen.

**Comment by emtudo** [comment]
Same problem: https://github.com/laravel/nova-issues/issues/21

**Comment by wongwangki** [comment]
this is a bug, the resource routing should follow the folder structure in app/Nova/
instead of using the class name (~file name) only
